examples/c-guest/Makefile

```diff
@@ -0,0 +1,52 @@
+ROOT_DIR:=$(strip $(dir $(realpath $(lastword $(MAKEFILE_LIST)))))
+
+ifeq ($(OS),Windows_NT)
+$(error Windows not supported)
+else
+    UNAME_S := $(shell uname -s)
+    ifeq ($(UNAME_S),Linux)
+        MACHINE = linux
+    endif
+    ifeq ($(UNAME_S),Darwin)
+        MACHINE = osx
+    endif
+
+    UNAME_P := $(shell uname -p)
+    ifeq ($(UNAME_P),x86_64)
+        MACHINE := $(MACHINE)-x86_64
+    endif
+    ifneq ($(filter arm%,$(UNAME_P)),)
+        MACHINE := $(MACHINE)-arm64
+    endif
+endif
+
+default: execute
+
+.PHONY: platform
+platform:
+	cargo +risc0 rustc -p zkvm-platform --target riscv32im-risc0-zkvm-elf --lib --crate-type staticlib --release --target-dir ./guest/out/platform
```

Comment:
I'm mostly writing this so I remember this in the future but I think this could be something we distribute on our end and it's something that can be downloaded by a command like `cargo risczero install`